plugins/streaming/impls/jms/src/main/java/org/pentaho/di/trans/step/jms/JmsStreamSource.java

```diff
@@ -92,7 +95,7 @@ private void receiveLoop() {
     //don't call super.close().  need to wait for the receiveLoop to be done
     if ( consumer != null && !closed.getAndSet( true ) ) {
       consumer.close();
-      jmsDelegate.getJmsContext().close();
+      jmsContext.close();
```

Comment:
1- If close is called before open, this will NPE
2- Since consumer = jmsContext (line 63), you're calling close twice on the same object

Reply:
Marc? You realize that when you move companies you can reset your to-do list, right? :-P

Not complaining at all, btw. Feel free to review all you like. :)